pkg/virtual/apiexport/controllers/apireconciler/apiexport_apireconciler_controller.go

```diff
@@ -121,17 +136,110 @@ type APIReconciler struct {
 	apiSets map[dynamiccontext.APIDomainKey]apidefinition.APIDefinitionSet
 }
 
+func (c *APIReconciler) gatherInternalAPIs() ([]internalapis.InternalAPI, error) {
+	_, apiResourcesLists, err := c.discoveryClient.ServerGroupsAndResources()
+	if err != nil {
+		return nil, err
+	}
+	internalAPIs := []internalapis.InternalAPI{}
+
+	for _, apiResourcesList := range apiResourcesLists {
+		gv, err := schema.ParseGroupVersion(apiResourcesList.GroupVersion)
+		if err != nil {
+			klog.Errorf("error parsing group version %v, err: %v", apiResourcesList.GroupVersion, err)
+			continue
+		}
+		// ignore kcp resources
+		if strings.HasSuffix(gv.Group, ".kcp.dev") {
+			continue
+		}
+		for _, apiResource := range apiResourcesList.APIResources {
+			// ignore subresources
+			if strings.Contains(apiResource.Name, "/") {
+				continue
+			}
+			resourceScope := apiextensionsv1.ClusterScoped
+			if apiResource.Namespaced {
+				resourceScope = apiextensionsv1.NamespaceScoped
+			}
+
+			gvk := schema.GroupVersionKind{Kind: apiResource.Kind, Version: gv.Version, Group: gv.Group}
+
+			instance, err := genericcontrolplanescheme.Scheme.New(gvk)
+			if err != nil {
+				if extensionsapiserver.Scheme.Recognizes(gvk) {
+					// Not currently supporting permission claim for CustomResourceDefinition.
+					// extensionsapiserver.Scheme is recursive and fails in internalapis.CreateAPIResourceSchemas in schema converter
+					klog.Warningf("permission claims not currently supported for gvk: %v/%v %v", gv.Group, gv.Version, apiResource.Kind)
+				} else {
+					klog.Errorf("error creating instance for gvk: %v/%v %v err: %v", gv.Group, gv.Version, apiResource.Kind, err)
+				}
+				continue
+			}
+
+			raw, err := runtime.DefaultUnstructuredConverter.ToUnstructured(instance)
+			if err != nil {
+				klog.Errorf("error checking status for gvk: %v/%v %v err: %v", gv.Group, gv.Version, apiResource.Kind, err)
+				continue
+			}
+			u := unstructured.Unstructured{Object: raw}
+			_, hasStatus, _ := unstructured.NestedStringMap(u.Object, "status")
```

Comment:
Instead of this, what if you maintained a separate `map` of maybe gvk to `internalapis.InternalAPI`, and check if `apiResource.Name` has a suffix of `/status`. If so, strip off `/status` to get the name, and then look up the entry in the map (creating if needed) and set `HasStatus=true`?

Reply:
This would only get the resources that have status subresources as having status. This might not be an issue, but I wanted to ask to make sure I understand.

Reply:
I don't believe it's possible to have status and not have the status subresource. Have you encountered otherwise?

Reply:
For internal API's I don't think so, for some CRDs, yeah, I have. it might not be a problem just making sure that I understand :)

Reply:
Technically that can happen, sure. The status subresource is totally opt-in. But clearly in practice nobody will use status without the subresource.